browser/ipfs/ipfs_tab_helper.cc

```diff
@@ -94,9 +94,16 @@ bool IPFSTabHelper::MaybeCreateForWebContents(
   return true;
 }
 
-void IPFSTabHelper::IPFSLinkResolved(const GURL& ipfs) {
+void IPFSTabHelper::XIPFSPathLinkResolved(const GURL& ipfs) {
   ipfs_resolved_url_ = ipfs;
-  if (pref_service_->GetBoolean(kIPFSAutoRedirectDNSLink)) {
+  UpdateLocationBar();
+}
+
+void IPFSTabHelper::DNSLinkResolved(const GURL& ipfs) {
+  ipfs_resolved_url_ = ipfs;
+  DCHECK(ipfs.is_empty() || ipfs.SchemeIs(kIPNSScheme));
```

Comment:
Why do we expect empty GURL here, don't we only call this function when resolved_url.is_valid() is true at line 133 below?
Also, if you're using a DCHECK, we shouldn't need to add !ipfs_resolved_url_.is_empty() condition below.
Maybe just `DCHECK(ipfs.is_valid() && ipfs.SchemeIs(kIPNSScheme));` here?

Reply:
Fixed